src/core_plugins/elasticsearch/lib/check_es_version.js

```diff
@@ -50,16 +54,20 @@ module.exports = function checkEsVersion(server, kibanaVersion) {
         ip: node.ip,
       }));
 
-      server.log(['warning'], {
-        tmpl: (
-          'You\'re running Kibana <%= kibanaVersion %> with some newer versions of ' +
-          'Elasticsearch. Update Kibana to the latest version to prevent compatibility issues: ' +
-          '<%= getHumanizedNodeNames(nodes).join(", ") %>'
-        ),
-        kibanaVersion,
-        getHumanizedNodeNames,
-        nodes: simplifiedNodes,
-      });
+      // Don't show the same warning over and over again.
+      const warningNodeNames = getHumanizedNodeNames(simplifiedNodes).join(', ');
+      if (lastWarnedNodesForServer.get(server) !== warningNodeNames) {
```

Comment:
Probably worth some sort of comment here about how multi-tenancy comes into play... if it's written in a way so that even I can understand it, then it's a foolproof comment. :)